Breaks infinite redirect loop when login via social media or unrecognised email

agnessa opened the pull request:
In relation to this:
https://www.pivotaltracker.com/story/show/137809425

In fact there were two conditions under which the app would enter an infinite redirect loop:
- user logged in successfully using an email address that is not registered in the CMS
- user logged in successfully with a social media account

In the first case the user is informed their account in the CMS might not be set up properly. In the latter, they are informed they should log in using their email (for that to happen, they need to log out first using the API gateway)

tiagojsag commented:
- I am testing the login with facebook. I see the warning saying "log in with your email address". But, when I click on "log out", I stay on the same page. Can we have this page with more information ("social network login is not supported on the cms" or something) + links that are useful to the user

- 
![image](https://cloud.githubusercontent.com/assets/973964/25121997/b304c9b2-2423-11e7-86bf-01e436a8240a.png) On a separate note, I think the styles here are buggy

agnessa commented:
Added a flag to allow the frontend to recognise that a user has no access to site, which happens even though their logged in (e.g. via social media account) and render the dropdown differently. Also, changed the text of the message.

sorodrigo commented:
This is ready for reviewing
